core/services/relay/evm/mercury/report_codec.go

```diff
@@ -0,0 +1,117 @@
+package mercury
+
+import (
+	"fmt"
+	"math/big"
+	"sort"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/smartcontractkit/libocr/offchainreporting2/reportingplugin/median"
+	"github.com/smartcontractkit/libocr/offchainreporting2/types"
+)
+
+// NOTE:
+// This report codec is a copy/paste of the original median evmreportcodec
+// here:
+// https://github.com/smartcontractkit/offchain-reporting/blob/master/lib/offchainreporting2/reportingplugin/median/evmreportcodec/reportcodec.go
+//
+// As a hack, blockNumber is substituted for juelsPerFeeCoin.
+//
+// Feed ID is a constant that comes from the jobspec and is added to the report also.
+//
+// A prettier implementation is dependent on the generic libocr median plugin,
+// here: https://github.com/smartcontractkit/offchain-reporting/pull/386/files
+
+var reportTypes = getReportTypes()
+
+func getReportTypes() abi.Arguments {
+	mustNewType := func(t string) abi.Type {
+		result, err := abi.NewType(t, "", []abi.ArgumentMarshaling{})
+		if err != nil {
+			panic(fmt.Sprintf("Unexpected error during abi.NewType: %s", err))
+		}
+		return result
+	}
+	return abi.Arguments([]abi.Argument{
```

Comment:
shouldn't we add `feedID` here for the unpack on the Mercury Server?

Reply:
I added it on my side as `{Name: "feedID", Type: mustNewType("bytes32")}` in order to read it and use it for the cache